### PR TITLE
Add WIF documentation

### PR DESCRIPTION
**Pull request by lock14**
Added documentation about using WIF with guardian.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -196,6 +196,28 @@ To make this process secure, Guardian should only be run in `internal` or
...
+configuration that can be used to set up guardian in a secure manner.
+
+The following [attribute mappings](https://cloud.google.com/iam/docs/workload-identity-federation#mapping) map claims from the [Github Actions JWT](https://token.actions.githubusercontent.com/.well-known/openid-configuration)
+to Google STS token attributes.
+  * `google.subject=assertion.sub`
+  * `attribute.actor=assertion.actor`
+  * `attribute.aud=assertion.aud`
+  * `attribute.repository_owner_id=assertion.repository_owner_id`
+  * `attribute.repository_id=assertion.repository_owner_id`
+  * `attribute.repository_visibility=assertion.repository_owner_id`
```
Comment (shankiyani) [inline README.md:211]: copy pasta on `assertion.repository_owner_id`?

Comment (lock14) [reply README.md]: yes, fixed.

File: README.md [hunk 2]
```diff
@@ -196,6 +196,28 @@ To make this process secure, Guardian should only be run in `internal` or
 but still ensures only the approved and merged workflow is being executed for
 the pull request.
 
+## Workload Identity Federation
+Terraform service accounts have elevated privileges, following [WIF Best Practices](https://cloud.google.com/iam/docs/best-practices-for-using-workload-identity-federation)
+is recommended when setting up guardian to use Workload Identity Federation. Below is an example attribute
+configuration that can be used to set up guardian in a secure manner.
+
+The following [attribute mappings](https://cloud.google.com/iam/docs/workload-identity-federation#mapping) map claims from the [Github Actions JWT](https://token.actions.githubusercontent.com/.well-known/openid-configuration)
```
Comment (sethvargo) [inline README.md:204]:
```suggestion
The following [attribute mappings](https://cloud.google.com/iam/docs/workload-identity-federation#mapping) map claims from the [GitHub Actions JWT](https://token.actions.githubusercontent.com/.well-known/openid-configuration)
```

File: README.md [hunk 3]
```diff
@@ -196,6 +196,28 @@ To make this process secure, Guardian should only be run in `internal` or
...
+  * `attribute.aud=assertion.aud`
+  * `attribute.repository_owner_id=assertion.repository_owner_id`
+  * `attribute.repository_id=assertion.repository_owner_id`
+  * `attribute.repository_visibility=assertion.repository_visibility`
+  * `attribute.workflow_ref=assertion.workflow_ref`
+
+The following [attribute conditions](https://cloud.google.com/iam/docs/workload-identity-federation#conditions) verify that the request is coming from your GitHub organization
+and repository as well as restricting access to only the guardian workflows that run on the main branch.
+  * `attribute.repository_owner_id=<your-github-org>` 
+  * `attribute.repository_id=<your-repository-id>`
```
Comment (sethvargo) [inline README.md:217]: Nit: document how to get the numeric IDs for these

File: README.md [hunk 4]
```diff
@@ -196,6 +196,42 @@ To make this process secure, Guardian should only be run in `internal` or
...
+  * `attribute.repository_id=assertion.repository_owner_id`
+  * `attribute.repository_visibility=assertion.repository_visibility`
+  * `attribute.workflow_ref=assertion.workflow_ref`
+
+The following [attribute conditions](https://cloud.google.com/iam/docs/workload-identity-federation#conditions) verify that the request is coming from your GitHub organization
+and repository as well as restricting access to only the guardian workflows that run on the main branch.
+  * `attribute.repository_owner_id=<your-repository-owner-id>` 
+  * `attribute.repository_id=<your-repository-id>`
+  * `attribute.repository_visibility != "public"`
+  * `attribute.workflow_ref in ["guardian-admin", "guardian-plan", "guardian-apply", "guardian-run"]`
```
Comment (verbanicm) [inline README.md:219]: I would change this to show the condition CEL expression like: 

```shell
attribute.repository_owner_id == "<your-repository-owner-id>" &&
attribute.repository_id == "<your-repository-id>" &&
attribute.repository_visibility != "public" &&
attribute.ref == "refs/heads/main" &&
attribute.workflow_ref in [
  "<your-repository-owner-name>/<your-repository-name>/.github/workflows/guardian-admin.yml@refs/heads/main",
  "<your-repository-owner-name>/<your-repository-name>/.github/workflows/guardian-apply.yml@refs/heads/main",
  "<your-repository-owner-name>/<your-repository-name>/.github/workflows/guardian-plan.yml@refs/heads/main",
  "<your-repository-owner-name>/<your-repository-name>/.github/workflows/guardian-run.yml@refs/heads/main",
]
```

File: README.md [hunk 5]
```diff
@@ -196,6 +196,42 @@ To make this process secure, Guardian should only be run in `internal` or
...
+
+The following [attribute mappings](https://cloud.google.com/iam/docs/workload-identity-federation#mapping) map claims from the [GitHub Actions JWT](https://token.actions.githubusercontent.com/.well-known/openid-configuration)
+to Google STS token attributes.
+  * `google.subject=assertion.sub`
+  * `attribute.actor=assertion.actor`
+  * `attribute.aud=assertion.aud`
+  * `attribute.repository_owner_id=assertion.repository_owner_id`
+  * `attribute.repository_id=assertion.repository_owner_id`
+  * `attribute.repository_visibility=assertion.repository_visibility`
+  * `attribute.workflow_ref=assertion.workflow_ref`
```
Comment (verbanicm) [inline README.md:212]: Added ref to support the expression below:

```shell
  * `google.subject=assertion.sub`
  * `attribute.actor=assertion.actor`
  * `attribute.aud=assertion.aud`
  * `attribute.ref=assertion.ref`
  * `attribute.repository_owner_id=assertion.repository_owner_id`
  * `attribute.repository_id=assertion.repository_owner_id`
  * `attribute.repository_visibility=assertion.repository_visibility`
  * `attribute.workflow_ref=assertion.workflow_ref`
```

**Review by sethvargo — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (lock14) [reply README.md]: added `attribute.ref=assertion.ref` to the list

File: README.md (hunk 4 above)
Comment (lock14) [reply README.md]: done.

**Review by verbanicm — APPROVED** [review]

**Pull request merged**